src/main/java/lotto/prize/LottoPrize.java

```diff
@@ -4,34 +4,32 @@
 
 import java.util.Arrays;
 import java.util.List;
-import java.util.Map;
 import java.util.function.Function;
 import java.util.stream.Collectors;
 
 public enum LottoPrize {
-    FIRST(6, Money.from(2_000_000_000)),
-    SECOND(5, Money.from(1_500_000)),
-    THIRD(4, Money.from(50_000)),
-    FOURTH(3, Money.from(5_000)),
-    NONE(0, Money.from(0));
-
-    private static final Map<Integer, LottoPrize> LOTTO_PRIZES = Arrays.stream(LottoPrize.values())
-            .collect(Collectors.toMap(LottoPrize::getMatchCount, Function.identity()));
+    FIRST(6, 2_000_000_000, (matchInfo) -> matchInfo.isCountEqual(6)),
+    SECOND(5, 30_000_000, (matchInfo) -> matchInfo.isCountEqual(5) && matchInfo.isBonusMatch()),
+    THIRD(5, 1_500_000, (matchInfo) -> matchInfo.isCountEqual(5) && !matchInfo.isBonusMatch()),
```

Comment:
```suggestion
    THIRD(5, 1_500_000, (matchInfo) -> matchInfo.isCountEqual(5) && matchInfo.nonBonusMatch()),
```

Objects.isNull과 Objects.nonNull을 제공하는 이유에 대해 고민해보아요

Reply:
기존에는 메소드 호출 후에 !를 붙여서 사용했었는데, 
객체의 값을 불러와서 가공 후 사용하는 것 보다는 non~~ 메소드 사용도 좋을거 같습니다. 앞으로 애용해볼게요 :)